src/main/java/com/google/sps/servlets/CreateGameServlet.java

```diff
@@ -38,11 +40,49 @@
 */
 @WebServlet("/create-game-data")
 public class CreateGameServlet extends HttpServlet {
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Gson gson = new Gson();
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.sendRedirect("/index.html");
+        if(!isInputValid(request)) {
+            System.out.println("Input json is not valid");
+            System.exit(1);
+        }
+        String gameID = IDGenerator.gen();
+        response.getWriter().println(gameID);
+    }
+
+    /**
+    * Checks that the values/sizes of numStages, numHints, stageSpawnLocations, stageStarterHints,
+    * hintLocations, and hintTexts all are consistent with one another.
+    * @param request the HttpServletRequest of the doPost.
+    * @return a boolean marking whether or not the request is consistent with itself.
+    */
+    boolean isInputValid(HttpServletRequest request) {
+        int numStages = getNumStages(request);
+        ArrayList<Integer> numHints = getNumHints(request);
+        ArrayList<Coordinates> stageSpawnLocations = getStageSpawnLocations(request);
+        ArrayList<String> stageStarterHints = getStageStarterHints(request);
+        if(numHints.size() != numStages) return false;
+        if(stageSpawnLocations.size() != numStages) return false;
+        if(stageStarterHints.size() != numStages) return false;
+
+        ArrayList<ArrayList<Coordinates>> hintLocations = getHintLocations(request);
+        ArrayList<ArrayList<String>> hintTexts = getHintTexts(request);
+        for(int i = 0; i < numStages; i++) {
+            if(hintLocations.get(i).size() != numHints.get(i)) return false;
+            if(hintTexts.get(i).size() != numHints.get(i)) return false;
+        }
+        return true;
+    }
+
+    /**
+    * Retrieves the userID of the creator of the game.
+    * @param request the HttpServletRequest of the doPost.
+    * @return a String representing the userID.
+    */
+    private String getUserID(HttpServletRequest request) {
```

Comment:
Here's the doc with the defined fields. It was hidden in the appendix of our design doc, but I plan to add an APIs section to make them more visible: https://docs.google.com/document/d/1w0ovXFHzRoE2ai9Xa8TgWKkI9-l5IAxb9nI2xyH7rRo/edit

Regarding the parseRequest function, I do think this makes sense and I think I will do that. But I still prefer keeping all of my other parsing functions that I can call within parseRequest, just because they split the work up nicely.